bundles/org.openhab.binding.yamahamusiccast/src/main/java/org/openhab/binding/yamahamusiccast/internal/YamahaMusiccastHandler.java

```diff
@@ -0,0 +1,1331 @@
+                            groupId = "";
+                        } else if (role.equals("none")) {
+                            groupId = generateGroupId();
+                        }
+                    }
+
+                    if (action.equals("unlink")) {
+                        json = "{\"group_id\":\"\"}";
+                        httpResponse = setClientInfo(this.host, json);
+                    } else if (action.equals("link")) {
+                        json = "{\"group_id\":\"" + groupId + "\", \"zone\":\"" + mclinkSetupZone
+                                + "\", \"type\":\"add\", \"client_list\":[\"" + this.host + "\"]}";
+                        logger.info("setServerInfo json: {}", json);
+                        httpResponse = setServerInfo(mclinkSetupServer, json);
+                        // All zones of Model are required for MC Link
+                        tmpString = "";
+                        for (int i = 1; i <= zoneNum; i++) {
+                            switch (i) {
+                                case 1:
+                                    tmpString = "\"main\"";
+                                    break;
+                                case 2:
+                                    tmpString = tmpString + ", \"zone2\"";
+                                    break;
+                                case 3:
+                                    tmpString = tmpString + ", \"zone3\"";
+                                    break;
+                                case 4:
+                                    tmpString = tmpString + ", \"zone4\"";
+                                    break;
+                            }
+                        }
+                        json = "{\"group_id\":\"" + groupId + "\", \"zone\":[" + tmpString + "]}";
+                        logger.info("setClientInfo json: {}", json);
+                        httpResponse = setClientInfo(this.host, json);
+                        httpResponse = startDistribution(mclinkSetupServer);
+                    }
+                    break;
+                case CHANNEL_UNLINKMCSERVER:
+                    if (command.equals(OnOffType.ON)) {
+                        json = "{\"group_id\":\"\"}";
+                        httpResponse = setServerInfo(this.host, json);
+                        updateState(channelUID, OnOffType.OFF);
+                    }
+                    break;
+                case CHANNEL_RECALLSCENE:
+                    recallScene(command.toString(), zone);
+                    break;
+                case CHANNEL_REPEAT:
+                    setRepeat(command.toString());
+                    break;
+                case CHANNEL_SHUFFLE:
+                    setShuffle(command.toString());
+                    break;
+            } // END Switch Channel
+        }
+    }
+
+    @Override
+    public void initialize() {
+        thingLabel = thing.getLabel();
+        logger.info("YXC - Start initializing! - {}", thingLabel);
```

Comment:
Can this message be replaced by using the debugger or by increasing the framework's log level? See point 4 https://www.openhab.org/docs/developer/guidelines.html#f-logging

Same for the others in this method.

Reply:
Done